allennlp/tests/semparse/domain_languages/quarel_language_test.py

```diff
@@ -0,0 +1,127 @@
+import pytest
+
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.semparse.domain_languages.quarel_language import QuaRel
+from allennlp.tests.semparse.domain_languages.domain_language_test import check_productions_match
+from allennlp.semparse import ExecutionError
+
+class QuaRelLanguageTest(AllenNlpTestCase):
+    def setUp(self):
+        super().setUp()
+        self.language = QuaRel()
+
+    def test_constant_logical_form(self):
+        assert self.language.execute('world2').number == 2
+        with pytest.raises(ExecutionError, match='Unrecognized constant'):
```

Comment:
This error is tested on the base class already, so there's no need to test it again here.  You can if you really want to, but you don't need to.  It's good to have a test that the language has the particular constants that you expect, though (although, those are already implicitly tested with the more complex logical forms below).